### ISSUE TITLE
Document or add V4 Yqlib.Get usage equivalent

### ISSUE THREAD

**Issue by mkatychev** [mikefarah/yq#640]
**Document or add example of using yqlib on preexisting yaml nodes.**

I was curious if there is a practical way to pass yaml nodes to `Yqlib` in the same way `Yqlib.Get` works:
```go
Yqlib.Get(rootNode *yaml.Node, path string) ([]*NodeContext, error)
```

Note:
- All the public methods/functions I have found expect a byte slice, string, or filepath
- There does not seem (at least in the time I've spent trying to replicate the functionality) to be a pre-decoded ingress for the path traversal flow by passing in a 

**Describe the solution you'd like**
Provide an example similar to the `Get` method that returns a `*CandidateNode`
**Describe alternatives you've considered**

**Additional context**
My current project uses pre-decoded `yaml.Node`s that it passes to yqlib v3, the flow doesn't seem to match for v4

**Comment by mikefarah** [comment]
I could add a new function to the all_once_evaluator, bsically it would be doing

```
func EvaluateNodes(expression string, inputCandidates *list.List) (*list.List, error) {
   node, err := treeCreator.ParsePath(expression)
   return = treeNavigator.GetMatchingNodes(input, node)
}
```

**Comment by mkatychev** [comment]
I created a PR https://github.com/mikefarah/yq/pull/657 to try and implement the suggestion.

**Comment by mikefarah** [comment]
Fixed in 4.3.2

Note that I ended up moving your code @mkatychev to the Evaluator interface

**Comment by mkatychev** [comment]
Awesome, thanks for the quick turnaround!
